casper-private-auction-installer/src/main.rs

```diff
@@ -112,59 +123,75 @@ pub fn get_entry_points() -> EntryPoints {
 #[no_mangle]
 pub extern "C" fn call() {
     let entry_points = get_entry_points();
-    let auction_named_keys = data::create_auction_named_keys();
-    let auction_desig: String = runtime::get_named_arg("name");
+    let auction_named_keys = create_auction_named_keys();
+    let contract_name: String = runtime::get_named_arg(NAME);
     let (auction_hash, _) = storage::new_locked_contract(
         entry_points,
         Some(auction_named_keys),
-        Some(format!("{}_{}", auction_desig, data::AUCTION_CONTRACT_HASH)),
-        Some(format!("{}_{}", auction_desig, data::AUCTION_ACCESS_TOKEN)),
+        Some(format!("{contract_name}_{AUCTION_PACKAGE_HASH}")),
+        Some(format!("{contract_name}_{AUCTION_ACCESS_TOKEN}")),
     );
     let auction_key = Key::Hash(auction_hash.value());
     runtime::put_key(
-        &format!("{}_auction_contract_hash", auction_desig),
+        &format!("{contract_name}_{AUCTION_CONTRACT_HASH}"),
         auction_key,
     );
     runtime::put_key(
-        &format!("{}_auction_contract_hash_wrapped", auction_desig),
+        &format!("{contract_name}_{AUCTION_CONTRACT_HASH}_{WRAPPED}"),
         storage::new_uref(auction_hash).into(),
     );
 
     // Create purse in the contract's context
-    runtime::call_contract::<()>(auction_hash, "init", runtime_args! {});
+    runtime::call_contract::<()>(auction_hash, INIT, runtime_args! {});
 
     // Hash of the NFT contract put up for auction
     let token_contract_hash = ContractPackageHash::new(
-        runtime::get_named_arg::<Key>(data::NFT_HASH)
+        runtime::get_named_arg::<Key>(NFT_HASH)
             .into_hash()
             .unwrap_or_revert_with(ApiError::User(200)),
     );
     // Transfer the NFT ownership to the auction
-    let token_ids = vec![runtime::get_named_arg::<String>(data::TOKEN_ID)];
-
-    let auction_contract_package_hash = runtime::get_key(&format!(
-        "{}_{}",
-        auction_desig,
-        data::AUCTION_CONTRACT_HASH
-    ))
-    .unwrap_or_revert_with(ApiError::User(201));
+    let token_id: String = runtime::get_named_arg::<String>(TOKEN_ID);
+    let token_hash = base16::encode_lower(&runtime::blake2b(&token_id));
```

Comment:
If it's Hash, it's already a byte encoding of a blake2b hash, this should just be token_id, the string

Reply:
Ok thanks I misunderstood that part, I thought it was a custom string the user would provide. Will remove.

Reply:
I don't make it work without this encoding. Getting a `InvalidTokenIdentifier = 28` 
Do you have an precise example of that key, it seems to be it's just a series of numbers but I can't make it work out of the box, not sure why but it is probably why I tried to encode it again from a random string. I would gladly move those encoding in tests but I don't have runtime.

  InvalidTokenIdentifier = 28,

Reply:
I'll try to take a closer look over the weekend

Reply:
Thanks, I removed it in https://github.com/CasperLabs/casper-private-auction/pull/23/commits/3a56e6d49aced55900577114ea7cc06b1fd87c86#diff-9135625ae4709b36720e878940c3b03ef51eeb47db01d4e51a2fe161c2b1aa75